db/schema.rb

```diff
@@ -164,4 +164,23 @@
     t.index ["uuid"], name: "index_teacher_clues_on_uuid", unique: true, using: :btree
   end
 
+
+  create_function :next_ecosystem_metadata_sequence_number, sql_definition: <<-SQL
+      CREATE OR REPLACE FUNCTION public.next_ecosystem_metadata_sequence_number()
+       RETURNS integer
+       LANGUAGE sql
+      AS $function$
+          SELECT PG_ADVISORY_XACT_LOCK(5676211225851683);
+          SELECT COALESCE(MAX("ecosystems"."metadata_sequence_number"), -1) + 1 FROM "ecosystems"
+        $function$
+  SQL
+  create_function :next_course_metadata_sequence_number, sql_definition: <<-SQL
+      CREATE OR REPLACE FUNCTION public.next_course_metadata_sequence_number()
+       RETURNS integer
+       LANGUAGE sql
+      AS $function$
+          SELECT PG_ADVISORY_XACT_LOCK(49102217655775016);
+          SELECT COALESCE(MAX("courses"."metadata_sequence_number"), -1) + 1 FROM "courses"
+        $function$
+  SQL
 end
```

Comment:
Just curious, but do you need to specify `.public` schema?   I thought that was the default and don't see it used very often, maybe this requires it for some reason though.

Reply:
Hm I think the fx gem adds that automatically, it isn't present in the .sql files that define the function.